packages/eslint-plugin/src/rules/no-deprecated-components/createNoDeprecatedComponentsRule.ts

```diff
@@ -137,6 +178,30 @@ export function createNoDeprecatedComponentsRule(
                             messageId: "migration",
                             node: node.property,
                         });
+                    } else if (isOpeningElement(node.parent)) {
+                        // check <Blueprint.DeprecatedComponent withDeprecatedProp={...}> syntax
+                        const deprecatedProp = node.parent.attributes.find(
+                            attribute =>
+                                attribute.type === TSESTree.AST_NODE_TYPES.JSXAttribute &&
+                                attribute.name.type === TSESTree.AST_NODE_TYPES.JSXIdentifier &&
+                                deprecatedToNewComponentMapping[`${node.property.name}.${attribute.name.name}`] != null,
+                        );
+                        if (deprecatedProp !== undefined) {
+                            const deprecatedComponentKey = Object.keys(deprecatedToNewComponentMapping).find(
+                                key => key.split(".")[0] === node.property.name,
+                            );
+                            context.report({
+                                data: {
+                                    deprecatedComponentName: node.property.name,
+                                    deprecatedPropName: (
+                                        (deprecatedProp as TSESTree.JSXAttribute).name as TSESTree.JSXIdentifier
+                                    ).name,
+                                    newComponentName: deprecatedToNewComponentMapping[deprecatedComponentKey!],
+                                },
+                                messageId: "migrationWithPropUsage",
+                                node,
+                            });
+                        }
```

Comment:
is this code repeated, just for a different node type?

Reply:
it's a little different, it uses `node.property.name` instead of `node.name`. but yes it's very similar to the other block. I think it would be a little convoluted / unreadable if we tried to DRY this code between the two syntax handlers.

Reply:
on second thought, I've refactored this bit into a function shared between the two handlers, it's still legible